x/concentrated-liquidity/swaps.go

```diff
@@ -113,19 +120,38 @@ var (
 // As a result, the range from the end of position one to the beginning of position
 // two has no liquidity and can be skipped.
 //
-// Returbs the spread factors accrued per unit of liquidity.
-func (ss *SwapState) updateSpreadRewardGrowthGlobal(spreadRewardChargeTotal osmomath.Dec) osmomath.Dec {
+// Returns the spread factors accrued per unit of liquidity.
+func (ss *SwapState) updateSpreadRewardGrowthGlobal(spreadRewardChargeTotal osmomath.Dec, scalingFactor osmomath.Dec) (osmomath.Dec, error) {
+	// Initialize spreadRewardChargeTotalScaled with the original unscaled value
+	spreadRewardChargeTotalScaled := spreadRewardChargeTotal
+
+	// Only scale if the scaling factor is not equal to oneDecScalingFactor
+	if !scalingFactor.Equal(oneDecScalingFactor) {
+		var err error
+		spreadRewardChargeTotalScaled, err = scaleUpTotalEmittedAmount(spreadRewardChargeTotal, scalingFactor)
+		if err != nil {
+			return osmomath.ZeroDec(), fmt.Errorf("failed to scale up spread reward charge: %w", err)
+		}
+	}
+
+	// Update global spread reward growth with the UNSCALED total
+	// We could scale this up, but we would need to scale it right back down at the end of the swap.
+	// The globalSpreadRewardGrowth is used to determine how much the user sends to the spread reward address,
+	// and is not persisted in the pool's tick accumulators.
 	ss.globalSpreadRewardGrowth = ss.globalSpreadRewardGrowth.Add(spreadRewardChargeTotal)
```

Comment:
Yeah this _looks_ fine since `globalSpreadRewardGrowth` is initialized to zero and only used to settle balances post-swap. It does break the boundary of "only operate in scaled space for swaps" but if it's an important performance improvement I think it's okay to keep

I was going to suggest adding a comment on the field in swap strategy proto but I see you've done that already!